cluster-autoscaler/cloudprovider/azure/azure_container_service_pool.go

```diff
@@ -78,12 +72,26 @@ func NewContainerServiceAgentPool(spec *dynamic.NodeGroupSpec, am *AzureManager)
 	return asg, nil
 }
 
-//GetPool is an internal function which figures out agentPoolProfile from the list based on the pool name provided in the --node parameter passed
+//GetAKSAgentPool is an internal function which figures out ManagedClusterAgentPoolProfile from the list based on the pool name provided in the --node parameter passed
+//to the autoscaler main
+func (agentPool *ContainerServiceAgentPool) GetAKSAgentPool(agentProfiles *[]containerservice.ManagedClusterAgentPoolProfile) (ret *containerservice.ManagedClusterAgentPoolProfile) {
+	for _, value := range *agentProfiles {
+		profileName := *value.Name
+		glog.V(5).Infof("AKS AgentPool profile name: %s", profileName)
+		if profileName == (agentPool.azureRef.Name) {
```

Comment:
Is it possible to move this to a common method since this is similar in case of AKS and ACS  ?

Reply:
Though they look same, the agentProfile is different struct for ACS and AKS. If we want to use the same logic, then we need a conversion between them, which is much more complicated.

Reply:
Ah.. yes.. I remember now ...:-)